edb/edgeql/compiler/__init__.py

```diff
@@ -519,7 +519,7 @@ def compile_func_to_ir(
 
     return_typemod = func.get_return_typemod(schema)
     if (return_typemod is not qltypes.TypeModifier.SET_OF
-            and ir.cardinality is qltypes.SchemaCardinality.MANY):
+            and ir.full_cardinality.is_multi()):
```

Comment:
Is there a reason why you want `cardinality` here? I'd rather have consistent naming for the two different cardinality enum attributes. As I explained in the PR blurb, "cardinality" is ultimately used as a property name in schema, so it makes sense to keep that name for things that map onto that, whereas any other notion of cardinality should have a clearly distinct name to avoid confusion.

Reply:
`full_cardinality` rubs me the wrong way, especially considering the inconsistency in the naming of the enum classes themselves.  Also, I don't think there's really that much room for confusion.  The IR and most of the compiler pipeline does not interact with schema cardinality directly, and all places in IR where you currently use `SchemaCardinality` can easily be replaced with `Cardinality`.

Reply:
I've renamed things back to "cardinality".